draft-ietf-quic-http.md

```diff
@@ -201,6 +201,21 @@ additional framing. Note that a request or response without a body will cause
 this stream to be half-closed in the corresponding direction without
 transferring data.
 
+Because the message control stream contains HPACK data which manipulates
+connection-level state, the message control stream MUST NOT be closed with a
+stream-level error.  If an implementation chooses to reject a request with a
+QUIC error code, it MUST trigger a RST_STREAM on the data stream only.  An
+implementation MAY close (FIN) a message control stream without completing a
+full HTTP message if the data stream has been abruptly closed.  Data on message
+control streams MUST be fully consumed, or the connection terminated.
+
+If a message control stream is terminated abruptly by the transport layer (e.g.
+with QUIC_TOO_MANY_OPEN_STREAMS), any frames already sent on that stream MUST
+be re-played on another message control stream as soon as the maximum number of
+streams permits.  If the implementation no longer requires the response to this
+request, it MAY abruptly terminate the corresponding data stream at the same
+time.
```

Comment:
@martinthomson I love the idea of always tearing down, but I'm a bit nervous about the interactions with stream limits and REFUSED streams. You mention 0-RTT in your comment. Is your idea that 0-RTT streams may need to be retried and not result in the connection being closed and might have to do a retry dance?